### DISCUSSION TITLE
blog/istio-service-mesh-book/

### DISCUSSION THREAD

**Discussion by giscus** [rootsongjc/rootsongjc.github.io#285] [Announcements]
# blog/istio-service-mesh-book/

三年磨一剑，云原生社区著《深入理解 Istio —— 云原生服务网格进阶实战》正式上市开售啦！

https://jimmysong.io/blog/istio-service-mesh-book/

**Comment by rootsongjc** [comment]
本书目前为预售阶段，可参与京东618活动，预计一周后有货。

**Reply by rootsongjc** [reply]
现在已经可以买到了，我已经收到书了。
![图书](https://tva1.sinaimg.cn/large/e6c9d24ely1h3k9u4f0ffj20xc0m80uj.jpg)

**Comment by lisper-inmove** [comment]
你好，图2.2好像缺失了。
我在 iPhone XS 的 Kindle上面看的。

**Reply by rootsongjc** [reply]
下面是图 2-2。另外目前这本书还没发行电子版吧，不知道你是在哪获得？

![图 2-2](https://tva1.sinaimg.cn/large/e6c9d24ely1h3jl9etwmij20qw0geq3r.jpg)

**Reply by lisper-inmove** [reply]
亚马逊上面。不过出版日期很奇怪，2022年7月。发不上来图片，你可以搜一下看看，我对比了封面，一模一样的

**Reply by lisper-inmove** [reply]
我发现 2-2的图片就在2-3的上面。真是瞎子。打扰了,

**Reply by rootsongjc** [reply]
有链接吗？我在跟出版社确认一下吧

**Reply by lisper-inmove** [reply]
https://www.amazon.cn/dp/B0B4NM2SJV/ref=sr_1_1?keywords=%E6%B7%B1%E5%85%A5%E7%90%86%E8%A7%A3+istio&qid=1656133197&sprefix=%2Caps%2C123&sr=8-1

**Reply by rootsongjc** [reply]
我确实不知道这本书已上架 kindle，不过话说 kindle 明年六月不就退出中国了吗，不过还好现在还可以买。

**Comment by stevenlii** [comment]
目录能给看一下吗

**Reply by rootsongjc** [reply]
以下为本书目录：

- 01  Service Mesh概述	1
- 1.1  Service Mesh基本概念	2
- 1.2  后Kubernetes时代的微服务	5
- 1.2.1  重要观点	6
- 1.2.2  Kubernetes与Service Mesh	6
- 1.2.3  kube-proxy组件	8
- 1.2.4  Kubernetes Ingress与Istio Gateway	8
- 1.2.5  xDS协议	9
- 1.2.6  Envoy	11
- 1.2.7  Istio Service Mesh	12
- 1.3  什么是Istio	12
- 1.3.1  为什么使用Istio	13
- 1.3.2  Istio的平台支持	13
- 1.4  本章小结	14
- 02  核心功能	15
- 2.1  流量控制	15
- 2.1.1  路由和流量转移	16
- 2.1.2  弹性功能	18
- 2.1.3  调试能力	18
- 2.1.4  实现流量控制的自定义资源	19
- 2.2  安全	20
- 2.2.1  认证	20
- 2.2.2  授权	21
- 2.3  可观察性	21
- 2.4  本章小结	23
- 03  架构解析	24
- 3.1  Istio的架构组成	24
- 3.2  Istio的设计目标	25
- 3.3  Istio的架构变迁	26
- 3.4  控制平面	27
- 3.4.1  Pilot	27
- 3.4.2  Citadel	37
- 3.5  数据平面	45
- 3.5.1  数据平面的概念	45
- 3.5.2  Sidecar 注入及透明流量劫持	47
- 3.5.3  Sidecar流量路由机制分析	56
- 3.5.4  Envoy	85
- 3.5.5  MOSN	92
- 3.6  本章小结	97
- 04  安装与部署	98
- 4.1  安装	98
- 4.1.1  环境准备	98
- 4.1.2  安装 Kubernetes集群	99
- 4.1.3  安装 Istio	99
- 4.2  升级	101
- 4.2.1  金丝雀升级	101
- 4.2.2  原地升级	102
- 4.3  Bookinfo 示例	103
- 4.3.1  环境准备	104
- 4.3.2  部署应用	104
- 4.3.3  启动应用服务	105
- 4.3.4  确定Ingress的IP和端口	106
- 4.3.5  集群外部访问应用	107
- 4.4  Katacoda实验环境	107
- 4.4.1  概念	107
- 4.4.2  使用Katacoda 学习	107
- 4.4.3  创建Katacoda课程	109
- 4.5  本章小结	115
- 05  流量控制	116
- 5.1  流量控制CRD	116
- 5.1.1  VirtualService	117
- 5.1.2  DestinationRule	118
- 5.1.3  Gateway	118
- 5.1.4  ServiceEntry	121
- 5.1.5  Sidecar	122
- 5.2  路由	122
- 5.2.1  VirtualService	123
- 5.2.2  路由规则	125
- 5.2.3  DestinationRule	127
- 5.2.4  Gateway	129
- 5.2.5  ServiceEntry	130
- 5.3  流量镜像	134
- 5.3.1  流量镜像能够做什么	134
- 5.3.2  流量镜像的实现原理	135
- 5.3.3  流量镜像的配置	137
- 5.3.4  实践	138
- 5.4  Ingress/Egress	148
- 5.4.1  Ingress	148
- 5.4.2  Egress	165
- 5.5  超时	181
- 5.6  重试	185
- 5.7  熔断	187
- 5.7.1  使用熔断功能	188
- 5.7.2  Istio 熔断实现	191
- 5.8  故障注入	193
- 5.8.1  HTTPFaultInjection.Abort	194
- 5.8.2  HTTPFaultInjection.Delay	194
- 5.9  本章小结	198
- 06  可观察性	199
- 6.1  指标监控	199
- 6.1.1  Prometheus	200
- 6.1.2  Prometheus配置解析	204
- 6.1.3  Prometheus-Istio指标	209
- 6.2  可视化	211
- 6.2.1  Grafana	212
- 6.2.2  Kiali	229
- 6.3  日志	243
- 6.3.1  传统日志	243
- 6.3.2  云原生日志	244
- 6.3.3  Istio日志	244
- 6.3.4  ELK	244
- 6.3.5  EFK	258
- 6.4  分布式追踪	272
- 6.4.1  Jaeger	273
- 6.4.2  Zipkin	279
- 6.4.3  Skywalking	285
- 6.5  本章小结	300
- 07  安全	301
- 7.1  认证	301
- 7.1.1  对等认证	303
- 7.1.2  请求认证	314
- 7.2  授权	319
- 7.2.1  授权策略	321
- 7.2.2  全局策略	322
- 7.2.3  局部策略	323
- 7.2.4  Match label	324
- 7.2.5  匹配算法	324
- 7.2.6  规则详解	325
- 7.2.7  操作示例	327
- 7.2.8  JWT 授权	332
- 7.3  本章小结	345
- 08  进阶实站	346
- 8.1  集成服务注册中心	346
- 8.1.1  Istio服务模型	346
- 8.1.2  Pilot服务模型源码分析	347
- 8.1.3  第三方服务注册表集成	349
- 8.2  对接API网关	351
- 8.2.1  Envoy	351
- 8.2.2  预备工作	352
- 8.2.3  开始监听	352
- 8.2.4  一条路由	354
- 8.2.5  一个服务	356
- 8.3  分布式追踪增强	360
- 8.3.1  OpenTracing	360
- 8.3.2  OpenTracing 概念模型	360
- 8.3.3  OpenTracing数据模型	361
- 8.3.4  跨进程调用信息传播	362
- 8.4  实现方法级的调用跟踪	364
- 8.4.1  Istio的分布式追踪	364
- 8.4.2  使用Opentracing来传递分布式跟踪上下文	367
- 8.4.3  在Istio中加入方法级的调用跟踪	368
- 8.5  实现Kafka消息跟踪	371
- 8.5.1  eshop 示例程序结构	372
- 8.5.2  将Kafka消息处理加入调用链跟踪	372
- 8.5.3  安装 Kafka集群	373
- 8.5.4  部署demo 应用	374
- 8.5.4  将调用跟踪上下文从Kafka 传递到REST服务	375
- 8.6  部署模型	379
- 8.6.1  集群模型 & 控制平面模型	379
- 8.6.2  网络模型	382
- 8.6.3  网格模型	382
- 8.6.4  身份和信任模型	383
- 8.6.5  租户模型	385
- 8.7  多集群部署与管理	385
- 8.7.1  多控制面	386
- 8.7.2  单控制面	391
- 8.8  本章小结	395
- 09  故障排查	396
- 9.1  常见使用问题	396
- 9.1.1  Service端口命名约束	397
- 9.1.2  流控规则下发顺序问题	398
- 9.1.3  请求中断分析	399
- 9.1.4  sidecar和user container启动顺序	402
- 9.1.5  Ingress Gateway和Service端口联动	403
- 9.1.6  VirtualService作用域	404
- 9.1.7  VirtualService不支持host fragment	405
- 9.1.8  全链路跟踪并非完全透明接入	406
- 9.1.9  mTLS 导致连接中断	407
- 9.2  诊断工具	408
- 9.2.1  istioctl命令行工具安装	409
- 9.2.2  使用proxy-status进行诊断	411
- 9.2.3  使用proxy-config命令进行诊断	414
- 9.2.4  使用analyze命令诊断	418
- 9.2.5  启用galley自动配置分析诊断	423
- 9.2.6  采用describe验证并理解网格配置	424
- 9.2.7  ControlZ自检工具	430
- 9.3  本章小结	432
- 10  Service Mesh生态	433
- 10.1  开源项目	433
- 10.1.1  Linkerd	433
- 10.1.2  Envoy	434
- 10.1.3  Istio	434
- 10.1.4  Consul connect	435
- 10.1.5  MOSN	435
- 10.1.6  Kong kuma	435
- 10.2  商业化项目	436
- 10.2.1  AWS	436
- 10.2.2  Google	436
- 10.2.3  Microsoft	437
- 10.2.4  Red Hat	437
- 10.2.4  Aspen Mesh	438
- 10.2.5  国内项目	438
- 10.3  标准	438
- 10.3.1  xDS	439
- 10.3.2  SMI	447
- 10.3.3  UDPA	451
- 10.4  扩展	452
- 10.4.1  WebAssembly	453
- 10.4.2  Contour	456
- 10.5  本章小结	472

**Comment by stevenlii** [comment]
这个讲的是1.5版本吗？

**Reply by rootsongjc** [reply]
是基于 Istio 1.11 的。
